### PR TITLE
Added a new input for the username field

### PR DESCRIPTION
**Pull request by CV2Tre**
Added a new input for the profile picture field
Users were not able to submit an account with a username shorter than 3 chars (onSubmit) Users were not able to create an account if the email or password were already in the database When valid data was submitted, the userContext was changed to the user that was just created, redirecting the user automatically to the home screen Ensured that the initialForm variable was also updated to reflect these changes

### CODE REVIEW

File: server/db/index.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ const mongoose = require('mongoose')
 require('dotenv').config()
 
 mongoose
-    .connect(process.env.MONGODB_CONNECTION_STRING)
+    .connect('mongodb://127.0.0.1:27017/travelBugDatabase')
```
Comment (AbeGue02) [inline server/db/index.js]: Change this back to what it was

File: server/controllers/UserController.js
Comment (AbeGue02) [inline server/controllers/UserController.js:1]: Test the changes and tell me if they work please

File: server/controllers/UserController.js [hunk 2]
```diff
@@ -53,11 +53,25 @@ const getUserByName = async (req, res) => {
 
 const createUser = async (req, res) => {
   try {
-    const user = await new User(req.body)
-    await User.save()
-    return res.status(201).json({user})
+    const { email, password } = req.body;
+    
+    const existingUser = await User.findOne({ $or: [{ email }, { password }] });
```
Comment (AbeGue02) [inline server/controllers/UserController.js]: This should be email or username. Multiple users can use the same password

File: server/controllers/UserController.js [hunk 3]
```diff
@@ -53,11 +53,25 @@ const getUserByName = async (req, res) => {
...
+
+    if (existingUser) {
+      return res.status(400).json({ error: 'Email or password already exists' });
```
Comment (AbeGue02) [inline server/controllers/UserController.js]: Again, email or username

File: travel-bug-app/src/components/CreateAccountScreen.jsx [hunk 4]
```diff
@@ -14,35 +15,76 @@ export default function CreateAccountScreen() {
...
 
-        // Change context for UserContext to update Main
+        if (formInfo.username.length < 3) {
+            alert("Username must be at least 3 characters long.")
+            return
+        }
 
-        setFormInfo(initialForm)
+        try {
+            const response = await fetch('http://localhost:3001/user', {
```
Comment (AbeGue02) [inline travel-bug-app/src/components/CreateAccountScreen.jsx]: use axios please

File: travel-bug-app/src/components/CreateAccountScreen.jsx [hunk 5]
```diff
@@ -1,8 +1,9 @@
-import { useState } from "react"
+import { useState } from "react";
```
Comment (AbeGue02) [inline travel-bug-app/src/components/CreateAccountScreen.jsx:5]: import axios

**Review by AbeGue02 — COMMENTED** [review]
Please review these changes before committing again

File: server/db/index.js (hunk 1 above)
Comment (CV2Tre) [reply server/db/index.js]: done

File: server/controllers/UserController.js (hunk 2 above)
Comment (CV2Tre) [reply server/controllers/UserController.js]: done

File: server/controllers/UserController.js (hunk 3 above)
Comment (CV2Tre) [reply server/controllers/UserController.js]: done

File: travel-bug-app/src/components/CreateAccountScreen.jsx (hunk 4 above)
Comment (CV2Tre) [reply travel-bug-app/src/components/CreateAccountScreen.jsx]: done

File: travel-bug-app/src/components/CreateAccountScreen.jsx (hunk 5 above)
Comment (CV2Tre) [reply travel-bug-app/src/components/CreateAccountScreen.jsx:5]: done

File: server/controllers/UserController.js
Comment (CV2Tre) [reply server/controllers/UserController.js:1]: done

**Review by AbeGue02 — APPROVED** [review]

**Pull request merged**